analysis/src/python/data_analysis/analysis/timing/sample_selection.py

```diff
@@ -0,0 +1,32 @@
+import argparse
+import sys
+from typing import List
+
+from analysis.src.python.data_analysis.model.column_name import SubmissionColumns, SubmissionStatsColumns
+from analysis.src.python.data_analysis.utils.df_utils import read_df
+
+
+def get_random_submission_with_lines_count(submissions_stats_path: str, lines_counts: List[int]) -> List[int]:
+    df_submissions_stats = read_df(submissions_stats_path)
+
+    submission_ids = []
+    for lines_count in lines_counts:
+        submission_stats = \
+            df_submissions_stats[df_submissions_stats[SubmissionStatsColumns.CODE_LINES_COUNT] == lines_count].sample()
+        submission_ids.append(submission_stats[SubmissionColumns.ID].values[0])
```

Comment:
Why do you use `values[0]`? Maybe is it better to use random fragments with this number of lines?
Also, we can group lines_counts by the same values and in this step gather several fragments by one time